romeo/src/proof_data.rs

```diff
@@ -1,24 +1,44 @@
 //! Proof Data used in Clarity Contracts
 use bdk::bitcoin::{Block, BlockHeader, Transaction, Txid as BitcoinTxId};
-use blockstack_lib::util::hash::{DoubleSha256, MerklePath, MerkleTree};
 use blockstack_lib::vm::types::{ListData, ListTypeData, SequenceData, Value, BUFF_32};
+use rs_merkle::{Hasher, MerkleTree};
+use sha2::{digest::FixedOutput, Digest, Sha256};
+
+/// The double sha256 algorithm used for bitcoin
+#[derive(Clone)]
+pub struct DoubleSha256Algorithm {}
+
+impl Hasher for DoubleSha256Algorithm {
+    type Hash = [u8; 32];
+
+    fn hash(data: &[u8]) -> [u8; 32] {
+        let mut hasher = Sha256::new();
+        let mut hasher2 = Sha256::new();
+        hasher.update(data);
+        hasher2.update(<[u8; 32]>::from(hasher.finalize_fixed()));
+        <[u8; 32]>::from(hasher2.finalize_fixed())
+    }
+}
 
 /// Data needed to prove that a bitcoin transaction was mined on the bitcoin
 /// network. This data is used by clarity contracts.
 #[derive(Debug, Clone)]
 pub struct ProofData {
-    /// The transaction id of the bitcoin transaction
-    pub txid: BitcoinTxId,
+    /// The reversed transaction id of the bitcoin transaction
```

Comment:
Can you document why we need the reversed transaction? Is this an issues with the Clarity Language?